lib/ansible/module_utils/facts.py

```diff
@@ -3031,15 +3034,35 @@ def get_file_lines(path):
     return ret
 
 def ansible_facts(module):
+    # Retrieve module parameters
+    gather_subset = "all"
+    if 'gather_subset' in module.params:
+        gather_subset = module.params['gather_subset']
+
+    # Retrieve all facts elements
+    if gather_subset == 'all':
+        gather_subset = 'min,hardware,network,virtual'
+
+    subsets = gather_subset.split(',')
```

Comment:
OK. Done in https://github.com/Yannig/ansible-modules-core/commit/3a4d62011400f394c3ae21fead307b6af8ded916